Add typeahead for tags via choices.js

Simrayz opened the pull request:
## Scope and purpose

Replaces #1146.

Replaces the tags input with a search multiselect, using `Choices.js`. The code now deals with lists of tags by default, instead of passing around a comma-separated list. To support the previous version and any saved bookmarks, the comma-separated tags format is supported on initial load, but is replaced in the URL with separated tags. 

The multiselect library was chosen because it was recommended as an alternative to Select2, but without a jQuery dependency. I initially implemented a version using Select2, but opted for not introducing any jQuery dependencies to this project just to implement a few filters.

I have added custom styles to the Choices select in order to more closely match the rest of Argus, as a new snippet and passed classes in the `Choices(...)` constructor. The choices initialization and event handlers are defined directly in the select template, as there is no existing method of defining "plugins" in Argus. We might want to look into how we can organize shared/"global" javascript.

## Screenshots

#### No selected tags and empty search bar
<img width="245" height="76" alt="image" src="https://github.com/user-attachments/assets/0dd29034-8929-48c0-bed4-d37f7eee1c9d" />

#### Searching for tags shows a list of choices fetched by the Choices instance from the `search_by_tags` view
<img width="252" height="157" alt="image" src="https://github.com/user-attachments/assets/7a7e3ad5-a053-44a6-8923-f831e07df7d3" />

#### Selecting a tag (or loading an URL with existing tags) will render as badges before the input field
<img width="379" height="69" alt="image" src="https://github.com/user-attachments/assets/e3ae20d4-d01b-40cd-854a-4060f8942bec" />

#### Selecting multiple tags will create a multi-line input.
This is not optimal visually, especially if users add *a lot* of tags. However, it is the only obvious way to both show selected tags, allow de-selection and include type-ahead.

<img width="419" height="96" alt="image" src="https://github.com/user-attachments/assets/065688fe-91e2-4135-a0b0-e759949aa6ae" />

## Contributor Checklist

Every pull request should have this checklist filled out, no matter how small it is.
More information about contributing to Argus can be found in the
[Development docs](https://argus-server.readthedocs.io/en/latest/development.html).

<!-- Add an "X" inside the brackets to confirm -->
<!-- Remove checks that do not apply -->
<!-- Of the checks that do apply: If not checking one or more of the boxes, please explain why below each. -->

* [x] Added a changelog fragment for [towncrier](https://argus-server.readthedocs.io/en/latest/development/howtos/changelog-entry.html)
* [ ] Added/amended tests for new/changed code
* [ ] Added/changed documentation, including updates to the [user manual](https://argus-server.readthedocs.io/en/latest/user-manual.html) if feature flow or UI is considerably changed
* [x] Linted/formatted the code with ruff and djLint, easiest by using [pre-commit](https://github.com/Uninett/Argus?tab=readme-ov-file#code-style)
* [x] The first line of the commit message continues the sentence "If applied, this commit will ...", starts with a capital letter, does not end with punctuation and is 50 characters or less long. See our [how-to](https://argus-server.readthedocs.io/en/latest/development/howtos/commit-messages.html)
* [ ] If applicable: Created new issues if this PR does not fix the issue completely/there is further work to be done
* [ ] If this results in changes in the UI: Added screenshots of the before and after
* [ ] If this results in changes to the database model: Updated the [ER diagram](https://argus-server.readthedocs.io/en/latest/development/howtos/regenerate-the-ER-diagram.html)

<!-- Make this a draft PR if the content is subject to change, cannot be merged or if it is for initial feedback -->

johannaengland reviewed (changes requested):
Before I start typing, when clicking into the field I get the message "No tags to choose from" in the dropdown. That makes me think there are no tags, I think it would be better to have something along the lines of "Begin typing to see tag suggestions", maybe even something about that two characters need to be entered before the search is done

And when typing in something I first get for a short moment a "No tags found" message before it shows me the results, so maybe a delay there is necessary

johannaengland commented on src/argus/htmx/templates/htmx/forms/search_select_multiple.html:
```diff
@@ -27,9 +28,9 @@
         loadingText: 'Loading...',
         noResultsText: `No ${parameterName} found`,
-        noChoicesText: `No ${parameterName} to choose from`,
+        noChoicesText: `Type at least ${minLength} times to see suggestions`,
```
```suggestion
        noChoicesText: `Enter at least ${minLength} characters to see suggestions`,
```

johannaengland reviewed (changes requested):
The problem is that now this message is also shown if no results are found after entering two characters

And I still get the `No tags found` message shortly before it shows me actual search results

Simrayz commented:
> If there is no way of differentiating between the message that is shown if no search text is entered yet and if no tags for that search are found, then I would prefer the "No tags found" message for all, because I think that is less confusing than the "Enter at least 2 characters" when two characters are already entered

@johannaengland I rebased the branch so it says "No tags to choose from" as before. I also added a new commit which should fix the short flash of "no results" when fetching from the backend.
